src/cursor.c

```diff
@@ -2103,11 +2103,16 @@ if (0) fprintf(stderr, "check_x11_pointer %d %d\n", root_x, root_y);
 		}
 	}
 
+	int x_rotated = x, y_rotated = y;
```

Comment:
You should internalise that rotation check inside cursor_position(). You're missing out on https://github.com/LibVNC/x11vnc/blob/a7564ed48a6fb88f4d831fe5abfced79313d2791/src/cursor.c#L2056 for instance.

Reply:
Yup, makes sense to me! I copied over the line from your PR and replaced the function with `rotate_cursor_coords`.